src/main/drivers/io_def.h

```diff
@@ -41,13 +41,23 @@
 // get ioRec by index
 #define DEFIO_REC_INDEXED(idx) (ioRecs + (idx))
 
+// TARGET must define used pins
```

Comment:
I'm not sure how to handle this. It is necessary to use preprocessor for DEFIO_TAG and friends to get compile-time constant. But only macros (macro names) should be public interface, implementation shall be hidden from user code. 

io.h had a bit different goals when implemented (target configuration was mostly defined at compile-time, io.h provided pin checking etc).

Only public guarantee is that `tag` (and `IO_t`) can be used in `if (tag)`. Even `unt8_t` size should be configurable. 

Maybe only forward macros in this file (`DEFIO_TAG(pinid) DEFIO_IMPL_TAG(pinid)`) and add new files (io_def_stm.h, io_def_pico.h).

Reply:
Perhaps common can cimply go into io.h, and then we move io_def.h and io_def_generated.h to the relevant platform directory.